src/traces/sankey/defaults.js

```diff
@@ -0,0 +1,136 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var d3 = require('d3');
+
+
+function linksDefaults(traceIn, traceOut) {
+    var linksIn = traceIn.links || [],
+        linksOut = traceOut.links = [];
+
+    var linkIn, linkOut, i;
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(linkIn, linkOut, attributes.links, attr, dflt);
+    }
+
+    for(i = 0; i < linksIn.length; i++) {
+        linkIn = linksIn[i];
+        linkOut = {};
+
+        if(!Lib.isPlainObject(linkIn)) {
+            continue;
+        }
+
+        var visible = coerce('visible');
+
+        if(visible) {
+            coerce('label');
+            coerce('value');
+            coerce('source');
+            coerce('target');
+            coerce('color');
+        }
+
+        linkOut._index = i;
+        linksOut.push(linkOut);
+    }
+}
+
+function nodesDefaults(traceIn, traceOut) {
+    var nodesIn = traceIn.nodes || [],
+        nodesOut = traceOut.nodes = [];
+
+    var nodeIn, nodeOut, i, j, link, foundUse, visible,
+        usedNodeCount = 0,
+        indexMap = [];
+
+    var defaultPalette = d3.scale.category20();
```

Comment:
It actually looks OK! A slight loss is that repetition is more noticeable as it has fewer colors, but users may specify per-node color and use a `d3` or `colorbrewer` categorical palette however they choose. I'll switch to the `plotly` palette for consistency.

With `d3.scale.category20`:
![image](https://cloud.githubusercontent.com/assets/1548516/25025910/70af2e4a-20a4-11e7-9a01-aec307a4953f.png)


With the 10 default `plotly.js` colors:
![image](https://cloud.githubusercontent.com/assets/1548516/25025855/3beb8f78-20a4-11e7-8d3e-cc7eb571a931.png)